.github/workflows/pytest.yml

```diff
@@ -0,0 +1,192 @@
+          run_id=$(echo "${{ inputs.upstream-details }}" | jq '".amalgam-lang-py"."run-id"')
+          gh $run_type download -D amlg -R "howsoai/amalgam-lang-py" -p "amalgam_lang-*-py3-none-any*" "$run_id"
+          # Needed because release/non-release downloads are different structure
+          cd amlg && if [ ! -f *.whl ]; then mv */*.whl ./; fi
+
+      - name: Download howso-validator-py
+        if: contains(inputs.upstream-details, 'howso-validator-py')
+        env:
+          GH_TOKEN: ${{ secrets.HOWSOAI_WORKFLOW_AUTOMATION_CLASSIC_TOKEN }}
+        run: |
+          run_type=$(echo "${{ inputs.upstream-details }}" | jq '".howso-validator-py"."run-type"')
+          run_id=$(echo "${{ inputs.upstream-details }}" | jq '".howso-validator-py"."run-id"')
+          gh $run_type download -D hsv -R "howsoai/howso-validator-py" -p "howso_validator-*-py3-none-any*" "$run_id"
+          # Needed because release/non-release downloads are different structure
+          cd hsv && if [ ! -f *.whl ]; then mv */*.whl ./; fi
+
+      - name: Download howso-validator-enterprise-py
+        if: contains(inputs.upstream-details, 'howso-validator-enterprise-py')
+        env:
+          GH_TOKEN: ${{ secrets.HOWSOAI_WORKFLOW_AUTOMATION_CLASSIC_TOKEN }}
+        run: |
+          run_type=$(echo "${{ inputs.upstream-details }}" | jq '".howso-validator-enterprise-py"."run-type"')
+          run_id=$(echo "${{ inputs.upstream-details }}" | jq '".howso-validator-enterprise-py"."run-id"')
+          gh $run_type download -D hsve -R "howsoai/howso-validator-enterprise-py" -p "howso_validator_enterprise-*-py3-none-any*" "$run_id"
+          # Needed because release/non-release downloads are different structure
+          cd hsve && if [ ! -f *.whl ]; then mv */*.whl ./; fi
+
+      - name: Download howso-synthesizer-py
+        if: contains(inputs.upstream-details, 'howso-synthesizer-py')
+        env:
+          GH_TOKEN: ${{ secrets.HOWSOAI_WORKFLOW_AUTOMATION_CLASSIC_TOKEN }}
+        run: |
+          run_type=$(echo "${{ inputs.upstream-details }}" | jq '".howso-synthesizer-py"."run-type"')
+          run_id=$(echo "${{ inputs.upstream-details }}" | jq '".howso-synthesizer-py"."run-id"')
+          gh $run_type download -D synth -R "howsoai/howso-synthesizer-py" -p "howso_synthesizer-*-py3-none-any*" "$run_id"
+          # Needed because release/non-release downloads are different structure
+          cd synth && if [ ! -f *.whl ]; then mv */*.whl ./; fi
+
+      - name: Configure MacOS environment
+        if: inputs.platform-pretty == 'MacOS'
+        run: |
+          brew install libomp
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python-version }}
+
+      - name: Download tz data
+        if: inputs.platform-pretty == 'Windows'
+        shell: pwsh
+        run: |
+          ./build/powershell/Download-Tzdata.ps1
```

Comment:
How is this one going to work on repos that don't have that file? Can we link scripts from this .github repo too? If not, we'll have to think about how to make the work with these new centralized wfs.

Reply:
Good point, I think this is true for `howso-engine-recipes`. I just put in a conditional to make sure it exists before calling it.